packages/server-wallet/src/channel-manager.ts

```diff
@@ -0,0 +1,93 @@
+import {CreateChannelParams} from '@statechannels/client-api-schema';
+
+import {MessageServiceInterface} from './message-service/types';
+import {getMessages} from './message-service/utils';
+import {WalletObjective} from './models/objective';
+import {Wallet} from './wallet';
+
+export const delay = async (ms = 10): Promise<void> =>
+  new Promise(resolve => setTimeout(resolve, ms));
+
+/**
+ * These are the default backoff intervals used.
+ * It uses a simple exponential strategy.
+ */
+export const DEFAULT_BACKOFF_INTERVALS = [1_000, 2_000, 4_000, 8_000, 16_000];
+
+export class ChannelManager {
+  /**
+   * Constructs a channel manager that will ensure objectives get accomplished by resending messages if needed.
+   * @param wallet The wallet to use.
+   * @param messageService  The message service to use.
+   * @param backOffIntervals At what intervals ensureObjective should retry sending messages to complete an objective.
+   * @returns A channel manager.
+   */
+  public static async create(
+    wallet: Wallet,
+    messageService: MessageServiceInterface,
+    backOffIntervals = DEFAULT_BACKOFF_INTERVALS
+  ): Promise<ChannelManager> {
+    return new ChannelManager(wallet, messageService, backOffIntervals);
+  }
+  private constructor(
+    private _wallet: Wallet,
+    private _messageService: MessageServiceInterface,
+    private _backoffIntervals: number[]
+  ) {}
+
+  /**
+   * TODO: This is a basic implementation of createChannels.
+   * This will be cleaned up in https://github.com/statechannels/statechannels/issues/3365
+   * @param args
+   * @param numberOfChannels
+   * @returns
+   */
+  public async createChannels(args: CreateChannelParams, numberOfChannels: number): Promise<void> {
```

Comment:
> * `createChannels` should enable the user to create channels that have different `CreateChannelParams`. so, could accept `CreateChannelParams[]`

Is that a more common case than creating a bunch of channels with the same `CreateChannelParams` ? My preference would be one of these:
- `CreateChannels(params:CreateChannelParams, numberOfChannels:number) `
- `CreateChannels(params:CreateChannelParams[])`

I don't think we should try to handle both cases with the same API call since that would make it more clunky to use.

Reply:
I think we should only handle `createChannels(params:CreateChannelParams[])`.
A user who wishes to call ```fancyCreateChannels(params:CreateChannelParams, numberOfChannels:number)``` can call ```createChannels(Array(n).fill(params))```